cas-server-support-osf/src/test/java/io/cos/cas/mock/MockOsfRemoteAuthenticateAction.java

```diff
@@ -0,0 +1,42 @@
+package io.cos.cas.mock;
+
+import io.cos.cas.authentication.OpenScienceFrameworkCredential;
+import io.cos.cas.authentication.handler.support.OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction;
+import org.jasig.cas.CentralAuthenticationService;
+
+import javax.security.auth.login.AccountException;
+
+/**
+ * This class mocks the {@link OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction} class.
+ *
+ * @author Longze Chen
+ * @since  4.1.5
+ */
+public class MockOsfRemoteAuthenticateAction
```

Comment:
Need to mock and extend the class `OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction` to override the method `notifyRemotePrincipalAuthenticated` that talks to OSF API.